test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriBuilder/UrlBuilderWithParameterAliasTests.cs

```diff
@@ -394,6 +394,46 @@ public void BuildExpandOrderby_AliasInFunction()
         }
         #endregion
 
+        #region alias in path segment
+        [Fact]
+        public void BuildFullUri_AliasInFilterPathSegment_AliasAsBoolean()
+        {
+            Uri fullUri = new Uri("http://gobbledygook/People/$filter=@p1?@p1=true");
+            ODataUriParser odataUriParser = new ODataUriParser(HardCodedTestModel.TestModel, serviceRoot, fullUri);
+            SetODataUriParserSettingsTo(this.settings, odataUriParser.Settings);
+            odataUriParser.UrlKeyDelimiter = ODataUrlKeyDelimiter.Parentheses;
+            ODataUri odataUri = odataUriParser.ParseUri();
+
+            odataUri.Filter.Should().BeNull();
+            odataUri.ParameterAliasNodes["@p1"].ShouldBeConstantQueryNode(true);
+
+            Uri actualUri = odataUri.BuildUri(ODataUrlKeyDelimiter.Parentheses);
+            Assert.Equal(fullUri, actualUri);
+
+            actualUri = odataUri.BuildUri(ODataUrlKeyDelimiter.Slash);
+            Assert.Equal(fullUri, actualUri);
+        }
+
+        [Fact]
+        public void BuildRelativeUri_AliasInFilterPathSegment_AliasAsBoolean()
+        {
+            Uri relativeUri = new Uri("People/$filter=@p1?@p1=true", UriKind.Relative);
+            ODataUriParser odataUriParser = new ODataUriParser(HardCodedTestModel.TestModel, serviceRoot, relativeUri);
+            SetODataUriParserSettingsTo(this.settings, odataUriParser.Settings);
+            odataUriParser.UrlKeyDelimiter = ODataUrlKeyDelimiter.Parentheses;
+            ODataUri odataUri = odataUriParser.ParseUri();
+
+            odataUri.Filter.Should().BeNull();

```

Comment:
>odataUri [](start = 12, length = 8)

do you want to add a check that the expected filter segment is present in the odataUri?

Reply:
:+1: